src/ast/passes/resource_analyser.cpp

```diff
@@ -243,6 +251,25 @@ void ResourceAnalyser::visit(Map &map)
   map_info.key_type = map.key_type;
 }
 
+void ResourceAnalyser::visit(Tuple &tuple)
+{
+  Visitor::visit(tuple);
+
+  resources_.tuple_buffers++;
+  resources_.max_tuple_size = std::max(resources_.max_tuple_size,
+                                       tuple.type.GetSize());
+}
+
+void ResourceAnalyser::visit(For &f)
+{
+  Visitor::visit(f);
+
+  // Need tuple per for loop to store key and value
```

Comment:
Does this handle the case where the key has a nested tuple?

Reply:
Yup! Because Visitor::visit(f) will recurse into the declared expression which will recurse into Visitor::visit(tuple)